File: 0443-string-compression/solution.py
class Solution(object):
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        totalLen = 0
        
        rIndex = 0
        wIndex = 0
        
        while rIndex < len(chars):
            curChar = chars[rIndex]
            count   = 0
            
            while rIndex < len(chars) and chars[rIndex] == curChar:
                rIndex += 1
                count  += 1
            
            if count == 1:
                chars[wIndex] = curChar
                wIndex += 1
                totalLen += 1
            else:
                chars[wIndex] = curChar
                wIndex += 1
                strCount = str(count)
                
                for i in xrange(len(strCount)):
                    chars[wIndex] = strCount[i]
                    wIndex += 1
        
                totalLen += 1 + len(strCount)
                
        return totalLen

File: 0443-string-compression/solution.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int ans = 0;
        const int n = chars.size();
        int prev = 0;
        for (int cur = 0; cur < n; ++cur, ++prev) {
            int numOccurances = 1;
            char curChar = chars[cur];
            while (cur + 1 < n && curChar == chars[cur+1]) {
                ++cur;
                ++numOccurances;
            }
            
            ++ans;
            

            if (prev != cur) {
                chars[prev] = curChar;
            }
            if(numOccurances > 1) {
                string numStr = to_string(numOccurances);
                ans += numStr.size();
                for (auto & c : numStr) {
                    chars[++prev] = c;
                }

                // chars[++prev] = curChar;


            }

        }

        return ans;
    }
};
